operator/src/main/java/org/bf2/operator/operands/KafkaCluster.java

```diff
@@ -666,38 +665,55 @@ private Storage buildKafkaStorage(ManagedKafka managedKafka, Kafka current) {
     /**
      * Get per broker value
      */
-    private long getPerBrokerBytes(ManagedKafka managedKafka, Kafka current, Quantity quantity, Supplier<String> defaultValue) {
+    private long getPerBrokerBytes(Quantity quantity, Supplier<String> defaultValue, int replicas) {
         long bytes = Quantity.getAmountInBytes(Optional.ofNullable(quantity).orElseGet(() -> Quantity.parse(defaultValue.get()))).longValue();
-
-        return bytes / getBrokerReplicas(managedKafka, current);
+        return bytes / replicas;
     }
 
     private long getIngressBytes(ManagedKafka managedKafka, Kafka current) {
-        return getPerBrokerBytes(managedKafka, current, managedKafka.getSpec().getCapacity().getIngressPerSec(), () -> this.configs.getConfig(managedKafka).getKafka().getIngressPerSec());
+        int brokerReplicas = getBrokerReplicas(managedKafka, current);
+        return getPerBrokerBytes(managedKafka.getSpec().getCapacity().getIngressPerSec(), () -> this.configs.getConfig(managedKafka).getKafka().getIngressPerSec(), brokerReplicas);
     }
 
     private long getEgressBytes(ManagedKafka managedKafka, Kafka current) {
-        return getPerBrokerBytes(managedKafka, current, managedKafka.getSpec().getCapacity().getEgressPerSec(), () -> this.configs.getConfig(managedKafka).getKafka().getEgressPerSec());
+        int brokerReplicas = getBrokerReplicas(managedKafka, current);
+        return getPerBrokerBytes(managedKafka.getSpec().getCapacity().getEgressPerSec(), () -> this.configs.getConfig(managedKafka).getKafka().getEgressPerSec(), brokerReplicas);
     }
 
     /**
-     * Get extra storage padding given the effective IngressEgressThroughput limit and storageMinMargin
+     * Calculates extra storage safety margin given the effective IngressEgressThroughput limit and storageMinMargin
      */
-    private long getStoragePadding(ManagedKafka managedKafka, Kafka current) {
+    private long calculateSafetyMargin(ManagedKafka managedKafka, Kafka current) {
         KafkaInstanceConfiguration config = this.configs.getConfig(managedKafka);
         return Quantity.getAmountInBytes(config.getStorage().getMinMargin()).longValue()
                 + getIngressBytes(managedKafka, current) * config.getStorage().getCheckInterval()
                         * config.getStorage().getSafetyFactor();
     }
 
+    enum FormattedValueType {
```

Comment:
Name feels clunky I'm wondering about `VolumeState` or even `VolumeFormatState`

Reply:
the enum is removed